File: backend/src/main/java/com/petcare/model/DietPlan.java
package com.petcare.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "diet_plan")
@Data
public class DietPlan {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "pet_id", nullable = false)
    private Long petId;
    
    @Column(name = "day_number", nullable = false)
    private Integer dayNumber;
    
    @Column(name = "meal_type", nullable = false)
    private String mealType; // breakfast, lunch, dinner
    
    @Column(columnDefinition = "TEXT", nullable = false)
    private String items;
    
    @Column(columnDefinition = "TEXT")
    private String addOns;
}

File: backend-python/app/models/diet_plan.py
from sqlalchemy import Column, Integer, String, Text
from app.database import Base

class DietPlan(Base):
    __tablename__ = "diet_plan"
    
    id = Column(Integer, primary_key=True, index=True)
    pet_id = Column(Integer, nullable=False, index=True)
    day_number = Column(Integer, nullable=False)
    meal_type = Column(String, nullable=False)
    items = Column(Text, nullable=False)
    add_ons = Column(Text)
